### PR TITLE
Push Libraries to newest Version

### PR DESCRIPTION
**Pull request by andiaa734**
RF24 to 1.4.2
PubSubClient to 2.8.0
CircularBuffer to 1.3.3
RichHttpServer to 2.0.3
ArduinoJson to 6.19.0
Platform to 2.3.6

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
 framework = arduino
-platform = espressif8266@~1.8
+platform = espressif8266@~2.6.3
 board_f_cpu = 160000000L
 lib_deps_builtin =
 lib_deps_external =
   WiFiManager=https://github.com/sidoh/WiFiManager.git#cmidgley
   RF24@~1.4.2
-  ArduinoJson@~6.10.1
+  ArduinoJson@~6.19.0
```
Comment (andiaa734) [inline platformio.ini:19]: ArduinoJson needs to be updated to 6.19.0 to work with Platform 2.6.3

File: lib/MiLight/MiLightClient.cpp [hunk 2]
```diff
@@ -322,7 +322,7 @@ void MiLightClient::update(JsonObject request) {
 
   JsonVariant brightness = request[GroupStateFieldNames::BRIGHTNESS];
   JsonVariant level = request[GroupStateFieldNames::LEVEL];
-  const bool isBrightnessDefined = !brightness.isUndefined() || !level.isUndefined();
+  const bool isBrightnessDefined = !brightness.isNull() || !level.isNull();
```
Comment (andiaa734) [inline lib/MiLight/MiLightClient.cpp:325]: isUndefined() was an unofficial Method in ArduinoJson and was removed in further Versions. So it needs to be replaced with isNull().

File: lib/MiLight/MiLightClient.cpp [hunk 3]
```diff
@@ -515,16 +515,16 @@ void MiLightClient::handleTransition(GroupStateField field, JsonVariant value, f
...
     return false;
   }
 
   const BulbId& bulbId = currentRemote->packetFormatter->currentBulbId();
-  const char* fieldName = args[FS(TransitionParams::FIELD)];
-  JsonVariant startValue = args[FS(TransitionParams::START_VALUE)];
-  JsonVariant endValue = args[FS(TransitionParams::END_VALUE)];
+  const char* fieldName = args[FSH(TransitionParams::FIELD)];
+  JsonVariant startValue = args[FSH(TransitionParams::START_VALUE)];
+  JsonVariant endValue = args[FSH(TransitionParams::END_VALUE)];
```
Comment (andiaa734) [inline lib/MiLight/MiLightClient.cpp:527]: FS was introduced in FS.h for Filesystem. So the Macro needed to be renamed.

**Comment by sidoh** [comment]
Thanks for doing this, @andiaa734! (and sorry it took me an eternity to get to reviewing) 

I updated all of the libraries in #802 for the v1.11.0 release so going to close this. There were actually a ton of issues that needed to be resolved to get things working with the latest SDK.

**Pull request closed without merging**